### PR TITLE
Show error toast when trying to download attachment when SD card is full

### PR DESCRIPTION
**Pull request by rhari991**
For #1476

### CODE REVIEW

**Comment by philipwhiuk** [comment]
Looks fine except remove references to SD card - we have no idea what they are trying to save to.

**Comment by rhari991** [comment]
Changed.

File: k9mail/src/main/java/com/fsck/k9/ui/messageview/AttachmentController.java [hunk 1]
```diff
@@ -123,6 +123,12 @@ private void saveAttachmentTo(File directory) {
             return;
         }
 
+        if (attachment.size > directory.getFreeSpace()) {
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/ui/messageview/AttachmentController.java]: I believe `attachment.size` contains size information from the header. That's not necessarily the actual file size.

**Comment by cketti** [comment]
Thanks!

**Pull request merged**